src/generic_core/diagnose-nat.ts

```diff
@@ -854,3 +856,310 @@ export function doNatProvoking() :Promise<string> {
         });
   });
 }
+
+// Test if NAT-PMP is supported by the router
+export function probePMPSupport(routerIP :string) :Promise<string> {
```

Comment:
I think it would make more sense for this function to return `Promise<boolean>` (`true` means supported) or `Promise<void>` (resolve means supported).